kube-runtime/src/events.rs

```diff
@@ -115,9 +158,10 @@ impl From<String> for Reporter {
 
 impl From<&str> for Reporter {
     fn from(es: &str) -> Self {
+        let instance = hostname::get().ok().and_then(|h| h.into_string().ok());
```

Comment:
had missed this earlier; how does this behave in-cluster in kubernetes? would this take the `node` name? presumably you are doing this for ambiguation of events in the case of multiple replicas (a not super common setup for controllers afaik).

it seems sensible enough to do, but maybe it's better to not do this implicitly and let people use the [downward api](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) to pass in the pod name instead of needing another dependency to get a less useful node name?

Reply:
Yes, I had the same concerns as you have. I decided to go with this because [hostname crate](https://crates.io/crates/hostname) is really small and I don't expect it to change, and also because `client-go` behaves in this way.

In Kubernetes, it takes the pod name as Hostname and this is "useful" because even with one replica if you recreate the pod you can differentiate them.

We can remove it from here, I don't have a strong opinion here. Furthermore, it is pretty easy to initiate it as you prefer in your controller.

Reply:
Nah, if it takes the pod name, then this is fantastic. Thanks a lot!